### PR TITLE
delete old metrics in puma collector

### PR DESCRIPTION
**Pull request by Fivell**
refs #69 #34

### CODE REVIEW

**Comment by Fivell** [comment]
image

**Comment by Fivell** [comment]
@SamSaffron  can't figure out why travis fails under 2.6 ruby , seems some technical problem

File: lib/prometheus_exporter/server/puma_collector.rb [hunk 1]
```diff
@@ -47,6 +48,15 @@ def metrics
     end
 
     def collect(obj)
+      now = ::Process.clock_gettime(::Process::CLOCK_MONOTONIC)
+
+      obj["created_at"] = now
+
+      @puma_metrics.delete_if do |current|
+        (obj["pid"] == current["pid"] && obj["hostname"] == current["hostname"]) ||
```
Comment (eviltrout) [inline lib/prometheus_exporter/server/puma_collector.rb]: I took a look at the puma collector and I don't think `pid` and `hostname` are included in the `obj`?

Comment (Fivell) [reply lib/prometheus_exporter/server/puma_collector.rb]: @eviltrout you're right , seems we need to leave only current["created_at"] + MAX_PUMA_METRIC_AGE < now

**Comment by eviltrout** [comment]
Thanks, I adapted this in e2d75a5aa84bc71e8bfcf07d8a406e1145ed9958 and credited you as a co-author.

**Pull request closed without merging**